libdeno/internal.h

```diff
@@ -56,8 +56,9 @@ class DenoIsolate {
   int32_t next_req_id_;
   void* user_data_;
 
-  // identity hash -> filename
-  std::map<int, std::string> module_filename_map_;
+  // identity hash -> filename, module (avoid hash collision)
+  std::multimap<int, std::pair<std::string, v8::Persistent<v8::Module>>>
+      module_info_map_;
   // filename -> Module
   std::map<std::string, v8::Persistent<v8::Module>> module_map_;
```

Comment:
I don't like that there's two `Persistent<Module>` references now. I don't understand why `module_filename_map_` was changed to include the `Persistent<Module>`...

Reply:
It is used to validate that the filename is indeed the one from the module: there might be multiple entries with the same hash but from different modules, and keeping the reference allows comparing if the identity matches (since we also have a reference to referrer)

Reply:
You haven't actually experience a hash collision yet? I would wait until that happens to introduce extra complexity. It seems likely that we will never experience it and due to the 
```
CHECK_EQ(0, module_filename_map_.count(id));
```
We'll see a crash if it happens, so it will be fairly obvious and easy to fix.

Reply:
Hmm... personally I prefer protective code, but yeah it is also an option to postpone until it really happens...

(Or at least we could leave a comment here about possible ways to address this issue if it ever happens...)

Reply:
Just had a discussion with TimothyGu (former Node TSC) about this, and he said that there were actually occasions when they observed IdentityHash collision, and it is never designed to be unique. He also suggested the Node way of keeping such persistent handles.

Reply:
Ok. Let's do it. I have a few minor issues but otherwise looks good.